src/extensions/extensions-api-client-internal.ts

```diff
@@ -14,26 +14,135 @@
-import { HttpClient, AuthorizedHttpClient } from '../utils/api-request';
-import { FirebaseAppError } from '../utils/error';
+import { AuthorizedHttpClient, HttpClient, HttpError, HttpRequestConfig } from '../utils/api-request';
+import { FirebaseAppError, PrefixedFirebaseError } from '../utils/error';
 import * as validator from '../utils/validator';
+import * as utils from '../utils';
+
+const FIREBASE_FUNCTIONS_CONFIG_HEADERS = {
+  'X-Firebase-Client': `fire-admin-node/${utils.getSdkVersion()}`
+};
+const EXTENSIONS_API_VERSION = 'v1beta';
+
 /**
  * Class that facilitates sending requests to the Firebase Extensions backend API.
  *
  * @internal
  */
 export class ExtensionsApiClient {
-  /* eslint-disable-next-line no-unused-vars */
   private readonly httpClient: HttpClient;
 
-  constructor(private readonly app: App) {
+  constructor(app: App) {
     if (!validator.isNonNullObject(app) || !('options' in app)) {
       throw new FirebaseAppError(
         'invalid-argument',
         'First argument passed to getExtensions() must be a valid Firebase app instance.');
     }
     this.httpClient = new AuthorizedHttpClient(app as FirebaseApp);
   }
+
+  async updateRuntimeData(
+    projectId: string,
+    instanceId: string,
+    runtimeData: RuntimeData
+  ): Promise<RuntimeDataResponse> {
+    const url = this.getRuntimeDataUri(projectId, instanceId);
+    const request: HttpRequestConfig = {
+      method: 'PATCH',
+      url,
+      headers: FIREBASE_FUNCTIONS_CONFIG_HEADERS,
+      data: runtimeData,
+    };
+    try {
+      const res = await this.httpClient.send(request);
+      return res.data
+    } catch (err: any) {
+      throw this.toFirebaseError(err);
+    }
+  }
+
+  private getExtensionsApiUri(): string {
+    return process.env['FIREBASE_EXT_URL'] ?? 'https://firebaseextensions.googleapis.com';
+  }
+
+  private getRuntimeDataUri(projectId: string, instanceId: string): string {
+    return `${
+      this.getExtensionsApiUri()
+    }/${EXTENSIONS_API_VERSION}/projects/${projectId}/instances/${instanceId}/runtimeData`;
+  }
+
+  private toFirebaseError(err: HttpError): PrefixedFirebaseError {
+    if (err instanceof PrefixedFirebaseError) {
+      return err;
+    }
+
+    const response = err.response;
+    if (!response?.isJson()) {
+      return new FirebaseExtensionsError(
+        'unknown-error',
+        `Unexpected response with status: ${response.status} and body: ${response.text}`);
+    }
+    const error = response.data?.error;
+    const message = error?.message || `Unknown server error: ${response.text}`;
+    switch (error.code) {
+    case 403:
+      return  new FirebaseExtensionsError('forbidden', message);
+    case 404:
+      return new FirebaseExtensionsError('not-found', message);
+    case 500:
+      return new FirebaseExtensionsError('internal-error', message);
+    }
+    return new FirebaseExtensionsError('unknown-error', message);
+  }
+}
+
+interface RuntimeData {
+
+  //oneof
```

Comment:
This is mainly for internal reference, since this type is only used internally. Also, I've never found a good way to accurately represent proto.OneOf in TS other than by just adding runtime assertions when using the type, so I usually just do this.